util/failover.go

```diff
@@ -129,169 +116,93 @@ func GetPod(clientset *kubernetes.Clientset, deploymentName, namespace string) (
 	return pod, err
 }
 
-// GetRepStatus is responsible for retrieving and returning the replication status of the pg replica
-// deployed in the pod specified.  Specifically, the function returns the location of the WAL file
-// that was most recently synced to disk on the replica (i.e. pg_last_xlog_receive_location() or
-// pg_last_wal_receive_lsn()), and the location of the WAL file that was most recently replayed on
-// the replica(i.e. pg_last_xlog_replay_location() or pg_last_wal_replay_lsn()), both as unint64.
-// Additionally, the  name of the Kubernetes node the replica is running on is returned.  This
-// function therefore provides insight into the replication lag for a replica, as needed to support
-// selection of replicas when performing manual failovers and scaling down the cluster.
-func GetRepStatus(restclient *rest.RESTClient, clientset *kubernetes.Clientset, pod *v1.Pod, namespace, databasePort string) (uint64, uint64, string, error) {
-	var err error
-
-	var receiveLocation, replayLocation uint64
-
-	var nodeName string
-
-	//get the crd for this dep
-	cluster := crv1.Pgcluster{}
-	var clusterfound bool
-	clusterfound, err = kubeapi.Getpgcluster(restclient, &cluster, pod.ObjectMeta.Labels[config.LABEL_PG_CLUSTER], namespace)
-	if err != nil || !clusterfound {
-		log.Error("Getpgcluster error: " + err.Error())
-		return receiveLocation, replayLocation, nodeName, err
+// ReplicationStatus is responsible for retrieving and returning the replication
+// information about the status of the replicas in a PostgreSQL cluster. It
+// executes into a single replica pod and leverages the functionality of Patroni
+// for getting the key metrics that are appropriate to help the user understand
+// the current state of their replicas.
+//
+// Statistics include: the current node the replica is on, if it is up, the
+// replication lag, etc.
+func ReplicationStatus(request ReplicationStatusRequest) (ReplicationStatusResponse, error) {
```

Comment:
I'm not sure how I feel about wrapping the 'client' handles in the request. I'm understanding that this is simply a util function, but it seems like the request should be 'data' only, IMV. :thinking: Based on what it's doing, seems like this could be it's own or part of a related 'service' and it would then have/be aware of it's own client state?

Reply:
@abrightwell I think this is something to think about for a future release. My goal was to take what we have in the existing infrastructure and ensure it works for the new HA changes. To have it work in that way would require a much larger effort compared to the remainder of the HA work that needs to take place.

That said, I could be convinced that this function could leave in the `apiserver` realm given it's called primarily by two `apiserver` endpoints, which would be a significantly lower effort to take place.

Reply:
> I think this is something to think about for a future release. My goal was to take what we have in the existing infrastructure and ensure it works for the new HA changes. To have it work in that way would require a much larger effort compared to the remainder of the HA work that needs to take place.

:+1: